### PR TITLE
Minimal coal store version

### PR DESCRIPTION
**Pull request by neomatamune**
- adds extra list of requirements for installing a fixed version of pandas 
- adds all list of requirements that add doc, test, and extra requirements
- Store comes with binding functions for:
    - csv
    - native python (list of dict)
    - pyarrow tables
    - pandas dataframes

### CODE REVIEW

File: cosmotech/coal/store/store.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        return column_name.replace(" ", "_").lower()
+
+    def __init__(self, reset=False):
+        self.store_location = pathlib.Path(os.environ.get("CSM_PARAMETERS_ABSOLUTE_PATH", ".")) / ".coal/store"
+        self.store_location.mkdir(parents=True, exist_ok=True)
+        self._tables = dict()
+        self._database_path = self.store_location / "db.sqlite"
+        if reset and self._database_path.exists():
+            self._database_path.unlink()
+        self._database = str(self._database_path)
```
Comment (lalepee) [inline cosmotech/coal/store/store.py:26]: Why not simply in a function name reset ?

File: cosmotech/coal/store/store.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    def get_table(self, table_name: str, columns: Optional[list[str]] = None) -> pyarrow.Table:
+        if not self.table_exists(table_name):
+            raise ValueError(f"No table with name {table_name} exists")
+        return self.execute_query(f"select * from {table_name}")
+        return pq.read_table(self._tables[table_name], columns=columns)
```
Comment (lalepee) [inline cosmotech/coal/store/store.py:32]: Double *return*, i'm rusty but this should not work

**Review by lalepee — APPROVED** [review]
Some changes, needed

File: cosmotech/coal/store/store.py (hunk 2 above)
Comment (neomatamune) [reply cosmotech/coal/store/store.py:32]: Forgot to remove the second return when I moved from parquet to sqlite

**Pull request merged**